torch/csrc/jit/codegen/cuda/index_reference_replay.cpp

```diff
@@ -39,6 +39,14 @@ void IndexReferenceReplay::handle(Split* s) {
   auto concrete_inner =
       GpuLower::current()->caIndexMap().getConcreteMappedID(s->inner());
 
+  if (concrete_outer->isParallelized()) {
+    replayed_outs.first->parallelize(concrete_outer->getParallelType());
```

Comment:
@naoyam I think we should check the parallel map here, is that correct?

Reply:
Do we also need to treat the pad option as a first class citizen of parallelization? i.e. build it into the compute at parallel map?

Reply:
@csarofeen I don't think that's necessary. The concrete axis should have the correct parallel type at this point. All axes that exist at the time of building the CA maps should get proper parallel types. That's not the case for reference axes as they don't exist when those maps are built.

Reply:
> Do we also need to treat the pad option as a first class citizen of parallelization? i.e. build it into the compute at parallel map?

I think we should reconsider how we represent things like padding and halo in a more consistent and robust way. I'm not sure what it should look like yet. I'm thinking that, once we get to the point that we have expressible enough constructs for generating optimized code, we should hopefully have more better ideas about what it should look like.